pkg/kn/commands/service/describe.go

```diff
@@ -99,10 +100,21 @@ func NewServiceDescribeCommand(p *commands.KnParams) *cobra.Command {
 		Example:           describe_example,
 		ValidArgsFunction: commands.ResourceNameCompletionFunc(p),
 		RunE: func(cmd *cobra.Command, args []string) error {
+			var serviceName string
 			if len(args) != 1 {
-				return errors.New("'service describe' requires the service name given as single argument")
+				if plugin.CtxManager != nil {
+					data, err := plugin.CtxManager.FetchContextData()
+					if err != nil {
+						return err
+					}
+					serviceName = data["service"]
+				}
+				if serviceName == "" {
+					return errors.New("'service describe' requires the service name given as single argument")
+				}
+			} else {
+				serviceName = args[0]
```

Comment:
I'm not sure, but I have the feeling that his access pattern will be repeated in several commands (all that require a service name), so maybe we could extract this handling in a utility method. Usually, I wouldn't opt for this, but this looks complex enough to avoid a C&P approach.